File: app/src/main/java/adapters/PerformanceAdapter.java
package adapters;

import android.app.Activity;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.sqube.tipshub.R;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Locale;
import java.util.Map;

public class PerformanceAdapter extends RecyclerView.Adapter<PerformanceAdapter.ListNewsViewHolder>{
    private Activity activity;
    private ArrayList<Map<String, Object>> performanceList;
    Map<String, Object> row = new HashMap<>();
    String TAG = "PerformanceAdapter";

    public PerformanceAdapter(Activity a, ArrayList<Map<String, Object>> list) {
        activity = a;
        performanceList = list;
        Log.i(TAG, "PerformanceAdapter: " + performanceList);
    }

    @NonNull
    @Override
    public ListNewsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Log.i("News started", "onCreateViewHolder: started");
        View convertView =  LayoutInflater.from(parent.getContext()).inflate(
                    R.layout.tips_performance_view, parent, false);
        return new ListNewsViewHolder(convertView);
    }


    @Override
    public void onBindViewHolder(@NonNull ListNewsViewHolder holder, int position) {
        row = performanceList.get(position);
        try{
            holder.txtType.setText(getType((int)row.get("type")));
            holder.txtNOG.setText((long)row.get("NOG")>1 ? row.get("NOG") + " tips" : row.get("NOG") + " tip");
            holder.txtWG.setText(row.get("WG")+ " won");
            long i = (long) row.get("WGP");
            holder.txtWGP.setText(String.format(Locale.getDefault(),"%.1f%%", (double) i));
        }catch(Exception e) {
            e.getStackTrace();
        }

    }

    @Override
    public int getItemCount() {
        Log.i(TAG, "getItemCount: " + performanceList.size());
        return performanceList.size();
    }

    private String getType(int i) {
        switch (i){
            case 1:
                return "3-5 odd";
            case 2:
                return "6-10 odd";
            case 3:
                return "11-50 odd";
            case 4:
                return "50+ odd";
            case 5:
                return "Draws";
            case 6:
                return "Banker";
        }
        return "";
    }

    class ListNewsViewHolder extends RecyclerView.ViewHolder {
        TextView txtType, txtNOG, txtWG, txtWGP;
        public ListNewsViewHolder(View itemView) {
            super(itemView);
            txtType = itemView.findViewById(R.id.txtType);
            txtNOG = itemView.findViewById(R.id.txtNOG);
            txtWG = itemView.findViewById(R.id.txtWG);
            txtWGP = itemView.findViewById(R.id.txtWGP);
        }
    }
}

File: app/src/main/java/adapters/PerformanceAdapter.kt
package adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.sqube.tipshub.databinding.ItemTipsPerformanceBinding
import java.util.*

class PerformanceAdapter(private val performanceList: ArrayList<Map<String, Any>>) : RecyclerView.Adapter<PerformanceAdapter.ListNewsViewHolder>() {
    private var row: Map<String, Any> = HashMap()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListNewsViewHolder {
        val binding = ItemTipsPerformanceBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ListNewsViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ListNewsViewHolder, position: Int) {
        row = performanceList[position]
        try {
            with(holder.binding){
                txtType.text = getType(row["type"] as Int)
                txtNOG.text = if (row["NOG"] as Long > 1) row["NOG"].toString() + " tips" else row["NOG"].toString() + " tip"
                txtWG.text = String.format(row["WG"].toString() + " won")
                val i = row["WGP"] as Long
                txtWGP.text = String.format(Locale.getDefault(), "%.1f%%", i.toDouble())
            }
        } catch (e: Exception) {
            e.stackTrace
        }
    }

    override fun getItemCount(): Int {
        return performanceList.size
    }

    private fun getType(i: Int): String {
        return when (i) {
            1 -> "3-5 odd"
            2 -> "6-10 odd"
            3 -> "11-50 odd"
            4 -> "50+ odd"
            5 -> "Draws"
            6 -> "Banker"
            else -> ""
        }
    }

    inner class ListNewsViewHolder(val binding: ItemTipsPerformanceBinding) : RecyclerView.ViewHolder(binding.root)
}